core/src/main/java/org/apache/brooklyn/util/core/xstream/XmlSerializer.java

```diff
@@ -80,6 +80,11 @@ protected MapperWrapper wrapMapper(MapperWrapper next) {
 
         xstream.registerConverter(new EnumCaseForgivingConverter());
         xstream.registerConverter(new Inet4AddressConverter());
+        
+        // See ObjectWithDefaultStringImplConverter (and its usage) for why we want to auto-detect 
+        // annotations (usages of this is in the camp project, so we can't just list it statically
+        // here unfortunately).
+        xstream.autodetectAnnotations(true);
```

Comment:
Seems to be about 5-10% overhead (when testing with a simple EntityMemento that has lots of string keys and brooklyn.parameters). That's certainly acceptable, to fix the issue!